wlauto/workloads/skypeecho/__init__.py

```diff
@@ -0,0 +1,143 @@
+
+SKYPE_ACTION_URIS = {
+    'call': 'call',
+    'video': 'call&video=true',
+}
+
+
+class SkypeEcho(AndroidUiAutoBenchmark):
+
+    name = 'skypeecho'
+    description = '''
+    Workload that makes Skype calls
+
+    It allows for the agenda to decide whether to make a voice call or a video call.
+    Credentials for the user account used to log into the Skype app have to be provided
+    in the agenda, as well as the display name and skype ID of the contact to call.
+
+    Other optional arguments allow controlling the duration of the call, whether the
+    call includes video or voice only, and whether to collect sys dumps.
+
+    For reliable testing, this workload requires a good and stable internet connection,
+    preferably on Wi-Fi.
+    '''
+    package = 'com.skype.raider'
+    activity = ''
+    # Skype has no default 'main' activity
+    launch_main = False # overrides extended class
+
+    instrumentation_log = '{}_instrumentation.log'.format(name)
+
+    parameters = [
+        Parameter('login_name', kind=str, mandatory=True,
+                  description='''
+                  Account to use when logging into the device from which the call will be made
+                  '''),
+        Parameter('login_pass', kind=str, mandatory=True,
+                  description='Password associated with the account to log into the device'),
+        Parameter('contact_skypeid', kind=str, mandatory=True,
+                  description='This is the Skype ID of the contact to call from the device'),
+        Parameter('contact_name', kind=str, mandatory=True,
+                  description='This is the contact display name as it appears in the people list'),
+        Parameter('duration', kind=int, default=60,
+                  description='This is the duration of the call in seconds'),
+        Parameter('action', kind=str, allowed_values=['voice', 'video'], default='video',
+                  description='Action to take - either video (default) or voice call'),
+        Parameter('use_gui', kind=bool, default=True,
+                  description='Specifies whether to use GUI or direct Skype URI'),
+        Parameter('dumpsys_enabled', kind=bool, default=True,
+                  description='''
+                  If ``True``, dumpsys captures will be carried out during the test run.
+                  The output is piped to log files which are then pulled from the phone.
+                  '''),
+    ]
+
+    def __init__(self, device, **kwargs):
+        super(SkypeEcho, self).__init__(device, **kwargs)
+        self.output_file = op.join(self.device.working_directory, self.instrumentation_log)
+        self.run_timeout = self.duration + 60
+
+    def validate(self):
+        super(SkypeEcho, self).validate()
+        self.uiauto_params['results_file'] = self.output_file
+        self.uiauto_params['dumpsys_enabled'] = self.dumpsys_enabled
+        self.uiauto_params['output_dir'] = self.device.working_directory
+        if self.use_gui:
```

Comment:
As above, if there is never a case when we would not 'use_gui' we should remove these.